### ISSUE TITLE
[Bug]: C++ error (unknown cause)

### ISSUE THREAD

**Issue by vicenteinefo** [jasp-stats/jasp-issues#2054]
### JASP Version

0.17.1

### JASP Module

ANOVA, Descriptives

### What analysis are you seeing the problem on?

Basic plots, Distribution plot, Scatter plots

### What OS are you seeing the problem on?

macOS Intel

### Bug Description

After applying a filter in a variable, the plots do not show and an error message is shown (i.e. C++ error (unknown cause))
<img width="1360" alt="Captura de pantalla 2023-03-13 a las 20 14 53" src="https://user-images.githubusercontent.com/29978768/224807498-b602d8d2-f56b-4d45-a957-59052f0fd24c.png">
[TFM data.jasp.zip](https://github.com/jasp-stats/jasp-issues/files/10960950/TFM.data.jasp.zip)

### Expected Behaviour

The plots should be shown with the data resulted of the applied filter

### Steps to Reproduce

1. Move right a scale variable
2. Execute any Distribution, Boxplot or scatter plot
3. Apply a filter to a nominal variable
4. Return to see the previous analysis done with the updated results
...

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [X] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by boutinb** [comment]
Hi @vicenteinefo We have tried this problem with different MacOS machines, but could not reproduce it. That makes it quite difficult us. Maybe you could also send us the log files of the engines: https://github.com/jasp-stats/jasp-desktop/blob/stable/Docs/user-guide/logging-howto.md

**Comment by vicenteinefo** [comment]
Hi @boutinb I have tried to save the log files following your instructions. Please, see the attached file. Hope this helps.
[Logs.zip](https://github.com/jasp-stats/jasp-issues/files/11222020/Logs.zip)

**Comment by boutinb** [comment]
@vandenman In the log files, I see 
```
Now rendering a plot with name: Con
could not find an old plot
Now rendering a plot with name: Exp
could not find an old plot
Now rendering a plot with name: Con
could not find an old plot
Now rendering a plot with name: Exp
could not find an old plot
Now rendering a plot with name: PACES total_Post1
could not find an old plot
Now rendering a plot with name: PACES total_Post2
could not find an old plot
Now rendering a plot with name: PACES total_Pre1
could not find an old plot
Now rendering a plot with name: PACES total_Pre2
could not find an old plot
Now rendering a plot with name: PACES total_Post
could not find an old plot
Now rendering a plot with name: PACES total_Pre
could not find an old plot
Now rendering a plot with name: Con
could not find an old plot
Now rendering a plot with name: Exp
could not find an old plot
Now rendering a plot with name: Con
could not find an old plot
Now rendering a plot with name: Exp
could not find an old plot
Now rendering a plot with name: PACES total_Pre - PACES total_Post
could not find an old plot
```
Does it ring a bell?

**Comment by vandenman** [comment]
Those messages are fine, they comes from here:

https://github.com/jasp-stats/jaspBase/blob/bac6409f03c548ecfccf2b25908678aa63e19666/src/jaspPlot.cpp#L88

and here:

https://github.com/jasp-stats/jaspBase/blob/bac6409f03c548ecfccf2b25908678aa63e19666/src/jaspPlot.cpp#L190

this is how we figure out if there was a previous plot where we could reuse the sizes (or plot editing information).

**Comment by boutinb** [comment]
@vicenteinefo Could you try then the last beta version of 0.17.2: https://static.jasp-stats.org/Nightlies/macOS-x86_64/JASP-nightly-stable-7ee09298bffe5bf602cb14133a7e557054b70c78.dmg ?

**Comment by vicenteinefo** [comment]
Hi @boutinb ! I have tried the beta version but the same error appears. Please, see attached below the log files in case you needed.
[JASP log 20230420.zip](https://github.com/jasp-stats/jasp-issues/files/11290387/JASP.log.20230420.zip)

**Comment by vandenman** [comment]
@boutinb I think the error originates in the R package `systemfonts`, see

https://github.com/r-lib/systemfonts/blob/99b6c630af663f382abe49cb8c596f482aa66661/src/utils.h#L33

so something with fonts might be wrong.

**Comment by boutinb** [comment]
Good catch: @vicenteinefo can you change your Preferences/Interface/Results fonts in JASP. Which one do you have? The default is Lucinda Grande.

**Comment by juliuspfadt** [comment]
@vicenteinefo, Your "Result & help" font is not Lucinda grande (at least in your screenshot). I can replicate your issue on macOS Intel when I have the settings as in your screenshot, changing the Result-help font to "default Lucinda grande" fixes the issue for me
![Screenshot 2023-04-21 at 11 24 01](https://user-images.githubusercontent.com/38500953/233599945-dc556a90-9c28-403d-9a7c-af4435b35b6f.png)

**Comment by vicenteinefo** [comment]
Great! This works for me also1 Thanks a lot.

**Comment by juliuspfadt** [comment]
any way we could get a more informative error message @vandenman?

**Comment by vandenman** [comment]
Well, we probably could, but it would be better if the fonts would just work on macOs :stuck_out_tongue:. It is not clear to me why this fails though.

**Comment by JorisGoosen** [comment]
Ill have to have a look and see if I can reproduce this on my intel mc

**Comment by tomtomme** [comment]
@vicenteinefo 
Can you test on macOS intel if changing a font to non-default does now work with jasp 0.18.3?

**Comment by JorisGoosen** [comment]
I think @boutinb fixed this by including some stuff from XQuartz in JASP.

This should be included in 0.18.3

**Comment by vicenteinefo** [comment]
> @vicenteinefo Can you test on macOS intel if changing a font to non-default does now work with jasp 0.18.3?

It seems it works!
